File: Jessica/14.Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        count=0
        length=len(strs)
        if  length==0:
            return ''
        if  length==1:
            return strs[0]
        len0=len(strs[0])
        for i in range(0,len0):
            for j in range(1,length):
                if i==len(strs[j]):
                    return strs[0][0:count]
                if strs[j][i] != strs[j-1][i]:
                    return strs[0][0:count]
                elif j==length-1:
                    count+=1
                else:                
                    continue            
        return strs[0][0:count]

File: Jessica/14.Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
       
        string ret="";
        int len=strs.size();
        if(len==0)
            return "";
        else if(len==1)        
            return strs[0];
        
        for(int i=0;i<strs[0].length();i++){
          
           for(int j=1;j<len;j++){
             
               if(strs[j][i] != strs[j-1][i])                   
                   return ret;                
               
               else{
                   
                   if(j==strs.size()-1)
                       ret+=strs[j][i];                     
                       
                                          
                   else
                       continue;
               }
               
               
           }
       }
           
        return ret;
    }
    
};